### PR TITLE
feat: [CDS-85163]: Created a new env variable for release notes

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
### Summary

Created a new environment variable for release notes and removed the dependency of deploymentType environment variable. 
The fallback is SAAS release notes link, so the setting of environment variable is required only in case of ON_PREM

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/10-common/components/ResourceCenter/utils.ts [hunk 1]
```diff
@@ -31,6 +31,8 @@ export const HARNESS_SUPPORT_LINK =
 export const CANNY_SUPPORT_LINK = 'https://ideas.harness.io/'
 export const WHATS_NEW_LINK = `https://docs.harness.io/article/ueeiji09do-what-s-new`
 export const EARLY_ACCESS_LINK = `https://docs.harness.io/article/w4krvu96i3-early-access`
+export const RELEASE_NOTES_LINK =
+  window.releaseNotesLink || 'https://ngdocs.harness.io/article/7zkchy5lhj-harness-saa-s-release-notes-2022'
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/ResourceCenter/utils.ts:35]: this is the new link (where the old one redirects): `https://developer.harness.io/release-notes/`
let's use this instead

File: src/modules/10-common/components/ResourceCenter/utils.ts [hunk 2]
```diff
@@ -80,22 +82,12 @@ export const cannySupportShareYourIdeas = (
   window.open(CANNY_SUPPORT_LINK)
 }
 
-export function getReleaseNodeLink(): string {
-  switch (window.deploymentType) {
-    case 'COMMUNITY':
-    case 'ON_PREM': {
-      return ON_PREM_RELEASE_NODE_LINK
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/ResourceCenter/utils.ts:87]: let's delete these two constants as well.. didnt eslint or TS complain about unused variables?

File: src/modules/10-common/components/ResourceCenter/__tests__/ResourceCenter.test.tsx [hunk 3]
```diff
@@ -158,20 +157,6 @@ describe('ResourceCenter', () => {
       })
     })
 
-    test('release note for on prem', async () => {
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/ResourceCenter/__tests__/ResourceCenter.test.tsx:161]: why remove the tests? we should update it.

Comment (avinashmadhwani02) [reply src/modules/10-common/components/ResourceCenter/__tests__/ResourceCenter.test.tsx:161]: added back

File: src/modules/10-common/components/ResourceCenter/utils.ts (hunk 2 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/components/ResourceCenter/utils.ts:87]: deleted SAAS url one. Using ON_PREM_RELEASE_NODE_LINK as fallback

File: src/global.d.ts [hunk 4]
```diff
@@ -91,6 +91,8 @@ declare interface Window {
   getIP?: (ipJson: { ip: string }) => void
   currentIP?: string | undefined
   publicAccessOnAccount?: boolean
+  releaseNotesLink: string
+  Canny: any
```
Comment (abhinavrastogi-harness) [inline src/global.d.ts:95]: why is Canny added in this PR? rebase issue?

File: src/modules/10-common/components/ResourceCenter/__tests__/ResourceCenter.test.tsx [hunk 5]
```diff
@@ -35,6 +34,9 @@ const zendeskCreate = {
   }
 }
 
+const SAAS_RELEASE_NODE_LINK = 'test_saas_link'
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/ResourceCenter/__tests__/ResourceCenter.test.tsx:37]: spelling.. NODE?

File: src/modules/10-common/components/ResourceCenter/__tests__/utils.test.ts [hunk 6]
```diff
@@ -41,4 +44,23 @@ describe('utility test', () => {
...
+  test('test default release notes link', () => {
+    expect(getReleaseNotesLink()).toEqual(ON_PREM_RELEASE_NODE_LINK)
+  })
+
+  test('test default release notes link', () => {
+    window.releaseNotesLink = 'test'
+    expect(getReleaseNotesLink()).toEqual('test')
+  })
+
+  test('test cannarySupportShareYourIdeas', () => {
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/ResourceCenter/__tests__/utils.test.ts:57]: why are tests for Canny included in this PR?
also, spelling.. `cannary`?

File: src/modules/10-common/components/ResourceCenter/utils.ts [hunk 7]
```diff
@@ -31,6 +30,7 @@ export const HARNESS_SUPPORT_LINK =
 export const CANNY_SUPPORT_LINK = 'https://ideas.harness.io/'
 export const WHATS_NEW_LINK = `https://docs.harness.io/article/ueeiji09do-what-s-new`
 export const EARLY_ACCESS_LINK = `https://docs.harness.io/article/w4krvu96i3-early-access`
+export const RELEASE_NOTES_LINK = window.releaseNotesLink || ON_PREM_RELEASE_NODE_LINK
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/ResourceCenter/utils.ts:33]: spelling.. `NODE`?

File: src/modules/10-common/components/ResourceCenter/utils.ts [hunk 8]
```diff
@@ -80,24 +80,18 @@ export const cannySupportShareYourIdeas = (
...
 }
+
 export const openEarlyAccess = (e: React.MouseEvent<Element, MouseEvent>): void => {
   e.stopPropagation()
   e.preventDefault()
   window.open(EARLY_ACCESS_LINK)
 }
+
+export const getReleaseNotesLink = () => {
+  return window.releaseNotesLink || ON_PREM_RELEASE_NODE_LINK
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/ResourceCenter/utils.ts:96]: logic duplicated from line 33. we only need one of these. i think we can remove this function, and just export/import the constant

File: src/modules/10-common/components/ResourceCenter/__tests__/utils.test.ts (hunk 6 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/components/ResourceCenter/__tests__/utils.test.ts:57]: Added to increase the coverage and avoid coverage check failure

File: src/modules/10-common/components/ResourceCenter/utils.ts (hunk 7 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/components/ResourceCenter/utils.ts]: Added to increase the coverage and avoid coverage check failure

File: src/modules/10-common/components/ResourceCenter/utils.ts (hunk 8 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/components/ResourceCenter/utils.ts]: removed line 33

File: src/modules/10-common/components/ResourceCenter/__tests__/utils.test.ts (hunk 6 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/components/ResourceCenter/__tests__/utils.test.ts]: added to increase the coverage of this file.

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by dhruvUpadhyay07 — APPROVED** [review]

**Pull request merged**